test/Dialect/FIRRTL/lower-intrinsics.mlir

```diff
@@ -55,13 +55,20 @@ firrtl.circuit "Foo" {
     firrtl.int.generic "circt_ltl_delay" <delay: i64 = 42> %in0 : (!firrtl.uint<1>) -> !firrtl.uint<1>
     firrtl.int.generic "circt_ltl_delay" <delay: i64 = 42, length: i64 = 1337> %in0 : (!firrtl.uint<1>) -> !firrtl.uint<1>
 
+    // CHECK-NEXT: firrtl.int.ltl.repeat %in0, 42 :
+    // CHECK-NEXT: firrtl.int.ltl.repeat %in0, 42, 1337 :
+    firrtl.int.generic "circt_ltl_repeat" <base: i64 = 42> %in0 : (!firrtl.uint<1>) -> !firrtl.uint<1>
+    firrtl.int.generic "circt_ltl_repeat" <base: i64 = 42, more: i64 = 1337> %in0 : (!firrtl.uint<1>) -> !firrtl.uint<1>
```

Comment:
The FIRRTLSpec might also need to be updated. I'm saying this, because the encoding of LTLIntrinsics is [currently being updated](https://github.com/chipsalliance/chisel/pull/4053) and that may or may not affect the way you encode this in FIRRTL.

Reply:
:+1: re:chisel PR and synchronizing any changes there since as you say they're actively being reworked (thanks!!).

Just a bit of information to add to what you're saying:

Specific intrinsics are not documented in the FIRRTL spec, but should be documented by the implementation -- for us that's here: https://github.com/llvm/circt/blob/main/docs/Dialects/FIRRTL/FIRRTLIntrinsics.md .
It would be good to have these (and the other LTL intrinsics) documented there in the future.

This test is using the `firrtl.int.generic` op which is what both intrinsic expressions and intmodules are encoded as (expressions parsed that way, intmodules via LowerIntmodules), so at least this part should be fine either way it's expressed from Chisel.

Reply:
Oh that's great to hear! Thanks for the specification @dtzSiFive , I'm no expert on how things get lowered from FIRRTL to the FIRRTL dialect, I just wanted to make sure my updates weren't breaking anything ^^"